src/main/java/org/isf/utils/file/FileTools.java

```diff
@@ -46,17 +46,17 @@
 	private static final String[] BINARY_UNITS = { "B", "M", "G" }; //Byte, Megabyte, Gigabyte 
 
 	private static final String[][] dateTimeFormats = new String[][] {
-			{ "yyyy-MM-dd", "\\d{4}-\\d{2}-\\d{2}" },
-			{ "dd-MM-yyyy", "\\d{2}-\\d{2}-\\d{4}" },
-			{ "dd-MM-yy", "\\d{2}-\\d{2}-\\d{2}" },
-			{ "dd/MM/yyyy", "\\d{2}/\\d{2}/\\d{4}" },
-			{ "dd/MM/yy", "\\d{2}/\\d{2}/\\d{2}" },
-			{ "yyyy-MM-dd HHmm", "\\d{4}-\\d{2}-\\d{2} \\d{4}" },
-			{ "yyyy-MM-dd HHmmss", "\\d{4}-\\d{2}-\\d{2} \\d{6}" },
-			{ "yyyy-MM-dd_HHmmss", "\\d{4}-\\d{2}-\\d{2}_\\d{6}" },
-			{ "dd-MM-yy_HHmm", "\\d{2}-\\d{2}-\\d{2}_\\d{4}" },
-			{ "yyyy-MM-dd HHmm", "\\d{4}-\\d{2}-\\d{2} \\d{4}" },
-			{ "yyyy-MM-dd_HHmm", "\\d{4}-\\d{2}-\\d{2}_\\d{4}" },
+			{ "yyyy-MM-dd", "\\d{4}-\\d{2}-\\d{2}\\z" },
+			{ "dd-MM-yyyy", "\\d{2}-\\d{2}-\\d{4}\\z" },
+			{ "dd-MM-yyyy HHmm", "\\d{2}-\\d{2}-\\d{4} \\d{4}\\z" },
+			{ "dd-MM-yy", "\\d{2}-\\d{2}-\\d{2}\\z" },
+			{ "dd/MM/yyyy", "\\d{2}/\\d{2}/\\d{4}\\z" },
+			{ "dd/MM/yy", "\\d{2}/\\d{2}/\\d{2}\\z" },
+			{ "yyyy-MM-dd HHmm", "\\d{4}-\\d{2}-\\d{2} \\d{4}\\z" },
+			{ "yyyy-MM-dd HHmmss", "\\d{4}-\\d{2}-\\d{2} \\d{6}\\z" },
+			{ "yyyy-MM-dd_HHmmss", "\\d{4}-\\d{2}-\\d{2}_\\d{6}\\z" },
+			{ "dd-MM-yy_HHmm", "\\d{2}-\\d{2}-\\d{2}_\\d{4}\\z" },
+			{ "yyyy-MM-dd_HHmm", "\\d{4}-\\d{2}-\\d{2}_\\d{4}\\z" },
```

Comment:
As I said in the comments I don't think it's a good idea because we cannot know where the date is placed in the filename (if as a suffix or a postfix) but maybe we can predict that no other digits [1-9] need to be found on a pattern's hips

Reply:
I made two changes:

1. Start a pattern at a word boundary; this allows the string to be in a larger context and eliminates things like "2021-12-22 1100" matching the string "dd-MM-yy" (which was real bad because it formatted the date so it wasn't even correct)
2. Only add the formatted date to the resulting array if it is not already in the list

Reply:
1. the problem is not solved, because if I'm not wrong, it would skip cases like "_09-03-2020_" or "02-05-2020_1122". Unfortunately the underscore is a very common character and it would be taken as “non-word characters”. `\w` stands for `[A-Za-z0-9_]`, which includes also digits, so also cannot apply... I think we have to exclude just digits manually, with something like `\b^[1-9].....\b^[1-9]` but I could be wrong!! 
2. that is good!

Reply:
1. sorry, I was totally wrong... found out that we could use the [look-around assertion](https://www.regular-expressions.info/lookaround.html) (match but not capture) so we could precede each regex with `(?<![1-9])` and end with `(?![1-9])`

make sense?

Reply:
I don't believe your solution works as if we try "2021-12-22" it matches:
````
{ "yyyy-MM-dd", "(?<![1-9])(\\d{4}-\\d{2}-\\d{2})(?![1-9])" },
{ "dd-MM-yy", "(?<![1-9])(\\d{2}-\\d{2}-\\d{2})(?![1-9])" },
````
which results in two dates:
````
  <[2021-12-22T00:00:00.000 (java.util.Date),
     2022-12-21T00:00:00.000 (java.util.Date)]>
````
Note that the second regex ignores the leading  "20" and treats it as "21-12-22".  So instead of 2021 the year is 2022.

Reply:
it doesn't work but it should... maybe is not supported by `java.util.regex` like warned on https://regexr.com/ ?

anyway just trying...

![image](https://user-images.githubusercontent.com/2938553/108720493-2d575200-7521-11eb-8eef-6f7837ac9d15.png)

Reply:
1. what about `(?<![0-9])` and end with `(?![0-9])` 😅

Test (using old `main()`):
```
7. 2021-12-22 1100 -> [Wed Dec 22 00:00:00 CET 2021, Wed Dec 22 11:00:00 CET 2021]
```

Reply:
Thanks; that last change seems to do the trick. 👍

Reply:
nice to hear that!!! thanks a lot!